am243x_evm/am2434/r5f0_0: TI ADC cleanup

natto1784 opened the pull request:
- Allow ADC to have any voltage reference instead of only internal reference
    
- Edit DT to reflect the change above

- Add pinctrl support for the TI ADC driver

-  Add pins for ADC0 channels

natto1784 commented:
resolved conflicts

JarmouniA reviewed (changes requested):
Split this into 3 commits: driver/binding, boards, samples

natto1784 commented:
thanks, changed

natto1784 commented on samples/drivers/adc/adc_sequence/boards/am243x_evm_am2434_r5f0_0.overlay:
Hi @JarmouniA, apologies for the late reply, this is the only platform that has an overlay for this. I have also rebased and updated the patch.

Also, offtopic but the new sample `adc_stream` also causes the similar issues to the one you fixed in the aforementioned PR, it fails if you have `adc0` nodelabel in your DTS but not `io-channels` in `zephyr,user`. Do you suggest we make the same changes to that sample's YAML or should  I add an overlay? Thanks.

JarmouniA replied:
Yes better to add dependency on `io-channels` in yaml file.

natto1784 committed:
boards: am243x_evm: add pinctrl nodes for main_adc0

Add the missing pins for ADC0 channels in the main domain.

Signed-off-by: Amneesh Singh <amneesh@ti.com>

natto1784 committed:
drivers: adc: adc_ti_am335x: add support for pinctrl

Support configuration of pins for ADC channels in the driver.

Signed-off-by: Amneesh Singh <amneesh@ti.com>

natto1784 committed:
boards: am243x_evm: change ADC channel ref voltage

Remove custom DT property ti,vrefp and change reference voltage from
INTERNAL to VDD_1 for all channels

Signed-off-by: Amneesh Singh <amneesh@ti.com>

natto1784 committed:
drivers: adc: ti_am335x: make internal reference optional

Allow using other references by removing the hard requirement on it being
internal. To be able to use internal voltage, the DT property ti,vrefp can
be used, otherwise it is ignored.

Signed-off-by: Amneesh Singh <amneesh@ti.com>

natto1784 commented on drivers/adc/adc_ti_am335x.c:
```diff
@@ -294,10 +296,19 @@ static int ti_adc_init(const struct device *dev)
 
 	DEVICE_MMIO_MAP(dev, K_MEM_CACHE_NONE);
 
+	regs = DEV_REGS(dev);
```
Oh yes, it is an unrelated stray change, I will separate it out in another commit.

natto1784 replied:
done

natto1784 committed:
drivers: adc: adc_ti_am335x: fix register MMIO mapping

Bind registers to the local variable in the ti_adc_init() function
after mapping the address space via DEVICE_MMIO_MAP.

Signed-off-by: Amneesh Singh <amneesh@ti.com>

natto1784 commented on boards/ti/am243x_evm/am243x_evm_am2434_r5f0_0.dts:
```diff
@@ -98,6 +98,9 @@
 &main_adc0 {
 	ti,fifo = <0>;
+	pinctrl-0 = <&main_adc0_ain0 &main_adc0_ain1 &main_adc0_ain2 &main_adc0_ain3
+		     &main_adc0_ain4 &main_adc0_ain5 &main_adc0_ain6 &main_adc0_ain7>;
```
Thanks, it is indeed cleaner, done!
